spec/controllers/articles_controller_spec.rb

```diff
@@ -0,0 +1,336 @@
+      expect(assigns(:article).attributes).to eq(article_a.dup.attributes)
+      expect(response).to have_http_status(:success)
+    end
+  end
+
+  describe "#create" do
+    it 'creates a new article' do
+      valid_attributes = article_a.attributes.except("id")
+      valid_attributes["name"] = "ABAB"
+      get :create, params: { foodcoop: FoodsoftConfig[:default_scope], supplier_id: supplier.id, article: valid_attributes }, xhr: true
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'fails to create a new article and renders #new' do
+      get :create, params: { foodcoop: FoodsoftConfig[:default_scope], supplier_id: supplier.id, article: { id: nil } }, xhr: true
+      expect(response).to have_http_status(:success)
+      expect(response).to render_template('articles/new')
+    end
+  end
+
+  describe "edit" do
+    it 'opens form to edit article attributes' do
+      get :edit, params: { foodcoop: FoodsoftConfig[:default_scope], supplier_id: supplier.id, id: article_a.id }, xhr: true
+      expect(response).to have_http_status(:success)
+      expect(response).to render_template('articles/new')
+    end
+  end
+
+  describe "#edit all" do
+    it 'renders edit_all' do
+      get :edit_all, params: { foodcoop: FoodsoftConfig[:default_scope], supplier_id: supplier.id }, xhr: true
+      expect(response).to have_http_status(:success)
+      expect(response).to render_template('articles/edit_all')
+    end
+  end
+
+  describe "#update" do
+    it 'updates article attributes' do
+      get :update, params: { foodcoop: FoodsoftConfig[:default_scope], supplier_id: supplier.id, id: article_a.id, article: { unit: "300 g" } }, xhr: true
+      expect(assigns(:article).unit).to eq("300 g")
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'updates article with empty name attribute' do
+      get :update, params: { foodcoop: FoodsoftConfig[:default_scope], supplier_id: supplier.id, id: article_a.id, article: { name: nil } }, xhr: true
+      expect(response).to render_template('articles/new')
+    end
+  end
+
+  describe "#update_all" do
+    it 'updates all articles' do
+      get :update_all, params: { foodcoop: FoodsoftConfig[:default_scope], supplier_id: supplier.id, articles: { "#{article_a.id}": attributes_for(:article), "#{article_b.id}": attributes_for(:article) } }
+      expect(response).to have_http_status(:redirect)
+    end
+
+    it 'fails on updating all articles' do
+      get :update_all, params: { foodcoop: FoodsoftConfig[:default_scope], supplier_id: supplier.id, articles: { "#{article_a.id}": attributes_for(:article, name: 'ab') } }
+      expect(response).to have_http_status(:success)
+      expect(response).to render_template('articles/edit_all')
+    end
+  end
+
+  describe "#update_selected" do
+    let(:order_article) { create :order_article, order: order, article: article_c }
+
+    before do
+      order_article
+    end
```

Comment:
i'd suggest to add a second order at line 15 instead. IMHO there should be no need for a dedicated `order_article` factory.